### PR TITLE
Add archiveSidecarsProofs to Database: persist KZG proofs and drop extension columns

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Another part of #10132 
No activation yet, just methods that will be called later + tests.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Medium Risk**
> Changes persisted blob/column storage and pruning behavior for Fulu data; incorrect completeness checks or prune/proof cleanup could make extension columns unrecoverable or leave stale proof entries.
> 
> **Overview**
> Adds **`archiveSidecarsProofs`** on `Database` / `KvStoreDatabase` to shrink Fulu storage: for each slot in a range with a **complete** first and extension half of data columns, it **stores extension-column KZG proofs**, **deletes extension sidecars**, and leaves the first-half columns for later reconstruction. Slots with any missing column in either half are **skipped** so archived data stays recoverable.
> 
> **Canonical `pruneAllSidecars`** now also **removes archived proofs** for pruned slots (with debug logging), keeping proof metadata aligned with sidecar pruning.
> 
> Includes integration coverage in `DatabaseTest`, serializer edge-case tests, and `CombinedChainDataClient` tests for reading stored proofs. **`NoOpDatabase`** stub added; wiring through the pruner is left for follow-up (not activated in this PR).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c2f53607e503b78904775a75ec3fad1c806644dc. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java [hunk 1]
```diff
@@ -1389,6 +1390,80 @@ public void pruneAllSidecars(final UInt64 tillSlotInclusive, final int pruneLimi
...
+              updater.addDataColumnSidecarsProofs(slot, proofs);
+              for (final DataColumnSlotAndIdentifier key : keys) {
+                updater.removeSidecar(key);
+              }
+              ++archivedSlots;
+              LOG.trace(
+                  "Pruned {} extension data column sidecars, keeping their proofs, at slot {}",
+                  keys.size(),
+                  slot);
+            }
```
[the remark this answers is not in the document]
Comment (zilm13) [reply storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java:1475]: fixed now

File: storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java [hunk 2]
```diff
@@ -1389,6 +1390,97 @@ public void pruneAllSidecars(final UInt64 tillSlotInclusive, final int pruneLimi
...
+                    .toList();
+            final long firstHalfCount =
+                archiveMap.get(slot).stream()
+                    .filter(id -> id.columnIndex().isLessThan(halfColumns))
+                    .count();
+
+            // Skip slots where either half is incomplete: first-half columns are required for
+            // reconstruction, and archiving extension columns without them would make the archived
+            // data irrecoverable.
+            if (extensionKeys.size() != halfColumns || firstHalfCount != halfColumns) {
```
[the remark this answers is not in the document]
Comment (zilm13) [reply storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java:1436]: the feature works only for finalized sidecars
it's impossible to finalize several roots for one slot
